.github/workflows/unit-and-integration-tests.yaml

```diff
@@ -23,18 +23,48 @@ jobs:
     needs:
       - call-inclusive-naming-check
 
-  integration-tests:
-    name: Integration tests on Testflinger instance with Intel GPU
-    runs-on: [self-hosted, testflinger]
+  integration-tests-system-A:
+    name: Integration tests A on Testflinger instance with Intel GPU(s)
+    runs-on: [testflinger]
     steps:
       - name: Check out code
         uses: actions/checkout@v4
-      - name: Substitute appropriate branch name in job definition file
+      - name: Build job file from template
+        env:
+          BRANCH: ${{ github.head_ref || github.ref_name }}
+          QUEUE: dell-precision-3470-c30322 #ADL iGPU + NVIDIA GPU
+          PROVISION_DATA: "distro: jammy"
         run: |
-          sed -i "s:REPLACE_BRANCH:${{ github.head_ref || github.ref_name }}:" \
-          ${GITHUB_WORKSPACE}/.github/testflinger/job-def.yaml
+          sed -e "s|REPLACE_BRANCH|${BRANCH}|" \
+          -e "s|REPLACE_QUEUE|${QUEUE}|" \
+          -e "s|REPLACE_PROVISION_DATA|${PROVISION_DATA}|" \
+          ${GITHUB_WORKSPACE}/.github/testflinger/job-def.yaml > \
+          ${GITHUB_WORKSPACE}/maas2.yaml
       - name: Submit testflinger job
         uses: canonical/testflinger/.github/actions/submit@main
         with:
           poll: true
-          job-path: ${GITHUB_WORKSPACE}/.github/testflinger/job-def.yaml
\ No newline at end of file
+          job-path: ${GITHUB_WORKSPACE}/maas2.yaml
```

Comment:
is it better to name this file after the nature of the job or platform instead of the deployment method ?

Reply:
My thought was that this may be helpful for future reference as it documents the provisioning method for a particular queue/system in testflinger. The filename is arbitrary, though, and isn't actually stored in the repo (it's just a temp file used for the final job submission, built from the template), so I don't mind changing it if you have a preference to change the naming. I did leave a comment after the queue name with some details about the GPU(s) on each system.